src/queue.rs

```diff
@@ -307,58 +300,63 @@ pub struct AvailIter<'b, M: GuestAddressSpace> {
     mem: M::T,
     desc_table: GuestAddress,
     avail_ring: GuestAddress,
-    next_index: Wrapping<u16>,
     last_index: Wrapping<u16>,
     queue_size: u16,
     next_avail: &'b mut Wrapping<u16>,
 }
 
-impl<'b, M: GuestAddressSpace> AvailIter<'b, M> {
-    /// Constructs an empty descriptor iterator.
-    pub fn new(mem: M::T, q_next_avail: &'b mut Wrapping<u16>) -> AvailIter<'b, M> {
-        AvailIter {
-            mem,
-            desc_table: GuestAddress(0),
-            avail_ring: GuestAddress(0),
-            next_index: Wrapping(0),
-            last_index: Wrapping(0),
-            queue_size: 0,
-            next_avail: q_next_avail,
-        }
-    }
-}
-
 impl<'b, M: GuestAddressSpace> Iterator for AvailIter<'b, M> {
     type Item = DescriptorChain<M>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        if self.next_index == self.last_index {
+        if *self.next_avail == self.last_index {
             return None;
         }
 
-        let offset = (VIRTQ_AVAIL_RING_HEADER_SIZE as u16
-            + (self.next_index.0 % self.queue_size) * VIRTQ_AVAIL_ELEMENT_SIZE as u16)
-            as usize;
-        let avail_addr = self.avail_ring.checked_add(offset as u64)?;
-        // This index is checked below in checked_new
-        let desc_index: u16 = self
+        // This computation cannot overflow.
```

Comment:
Mind to also say why it cannot overflow? :)) Otherwise, we can just remove the comment.

Reply:
Added a bit more details to the comment.